modules/file-system/gke-persistent-volume/main.tf

```diff
@@ -122,17 +154,17 @@ resource "kubectl_manifest" "pvc_namespace" {
 }
 
 resource "kubectl_manifest" "pv" {
-  yaml_body = local.is_gcs ? local.gcs_pv_contents : local.filestore_pv_contents
+  yaml_body = local.is_gcs ? local.gcs_pv_contents : (local.is_lustre ? local.lustre_pv_contents : local.filestore_pv_contents)
```

Comment:
These type of ternary conditional expression (including the previous ones) becomes unmaintanable as we add more types. Instead try a map lookup

```
storage_map = {
  gcs = is_gcs
  lustre = is_lustre
  ...
}

gcs_pv = ...
lustre_pv = ...
filrestore_pv = ...

pv_map = {
  gcs = local.gcs_pv
  lustre = local.lustre_pv
  ...
}
selected_storage = one([
    for key, value in local.storage_map : key if value != null
  ])

pv_yaml = local.pv_map[selected_storage]

```

You can modify this code for references in othe rplaces